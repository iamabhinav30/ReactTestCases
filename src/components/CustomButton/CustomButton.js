
import React from 'react';
import PropTypes from 'prop-types';

// import './CustomButton.css'

const CustomButton = ({btnLabel, clickHandler}) => {
  return <button type="button" name="action-btn" className="custom-btn" 
                               onClick={clickHandler}>{btnLabel}</button>;
}

CustomButton.propTypes = {
    btnLabel: PropTypes.string.isRequired,
    clickHandler: PropTypes.func.isRequired,
};

export default CustomButton;