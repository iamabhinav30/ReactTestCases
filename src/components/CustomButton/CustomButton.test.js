
import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon' ;

import CustomButton from "./CustomButton";

describe("CustomButton component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <CustomButton btnLabel={"Update"} clickHandler={jest.fn()} />
    );
  
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setNewCityName on click", () => {
    const spy = sinon.spy();

    const wrapper = mount(
      <CustomButton btnLabel={"Update"} clickHandler={spy} />
    );
    wrapper
      .find("button")
      .first()
      .simulate("click");
      
    expect(spy.calledOnce).toBe(true);
  });
});