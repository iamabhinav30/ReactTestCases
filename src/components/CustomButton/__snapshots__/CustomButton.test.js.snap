// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component renders correctly 1`] = `ShallowWrapper {}`;
